pkg/cli/run.go

```diff
@@ -101,8 +109,21 @@ func NewRun(c CommandContext) *cobra.Command {
 	if err := cmd.RegisterFlagCompletionFunc("region", newCompletion(c.ClientFactory, regionsCompletion).complete); err != nil {
 		cmd.Printf("Error registering completion function for --region flag: %v\n", err)
 	}
-	cmd.PersistentFlags().Lookup("dangerous").Hidden = true
+	toggleHiddenFlags(cmd, []string{"dangerous", "memory", "target-namespace", "secret", "volume", "region", "publish-all",
+		"publish", "link", "label", "interval", "image", "env", "compute-class", "annotation", "dev", "bidirectional-sync"}, true)
+
 	cmd.Flags().SetInterspersed(false)
+	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
+		fmt.Println(cmd.Short + "\n")
+		//toggle advanced flags on before printing flags out in cmd.UsageString
+		if slices.Contains(args, "--advanced") {
+			toggleHiddenFlags(cmd, []string{"dangerous", "memory", "target-namespace", "secret", "volume", "region", "publish-all",
+				"publish", "link", "label", "interval", "image", "env", "compute-class", "annotation", "dev", "bidirectional-sync"}, false)
+			//need to defer the print of advanced help message to print after usage string for formatting reasons
+			defer fmt.Println(cmd.Long)
```

Comment:
Neat trick! Also NIT: I like white space between // and the comment, but personal preference.